### PR TITLE
Feature/crawl leetcode

### PR DESCRIPTION
**Pull request by ghost**
inference when input is URL from leetcode

### CODE REVIEW

File: run_inference.py [hunk 1]
```diff
@@ -28,4 +62,6 @@ def main(input_file: str, model_path: str = typer.Option("khaimaitien/leetcode_s
 
 
 if __name__ == "__main__":
-    typer.run(main)
+    # typer.run(main)
+    reader = crawl_desc_leetcode_url("https://aaa.com/problems/substring-with-concatenation-of-all-words/")
```
Comment (khaimt) [inline run_inference.py]: Can you give the real link? and allow user to pass real-link in command line?
give 2 options: read from file and read from url
--url https://leetcode.com/problems/two-sum/
or 
--file test_cases/problem1.txt

python run_inference.py --url https://leetcode.com/problems/two-sum/

Comment (khaimt) [reply run_inference.py]: Also please update the readme

Comment (ghost) [reply run_inference.py]: @khaimt Thanks, I made changes, help me merge request!

File: run_inference.py [hunk 2]
```diff
@@ -1,31 +1,92 @@
...
+        if meta_description:
+            content_value = meta_description.get("content")
+            return content_value
+        else:
+            return None
+    else:
+        return None
+
+
+def main(leetcode_url: str = None, input_file: str = None, model_path: str = "khaimaitien/leetcode_solver_7b"):
```
Comment (khaimt) [inline run_inference.py]: Can you change this to:
leetcode_url: str = typer.Option("")
input_file: str = typer.Option("")

File: run_inference.py [hunk 3]
```diff
@@ -1,31 +1,92 @@
...
     prompt_template = get_prompt_template_by_name("code_llama")
+    algo_input = None
+    if input_file is not None:
+        # read the leetcode problem from input_file
+        with open(input_file, "r") as f:
+            algo_input = f.read()
 
-    tokenizer = AutoTokenizer.from_pretrained(model_path)
-    model = AutoModelForCausalLM.from_pretrained(model_path, device_map="auto", torch_dtype=torch.bfloat16)
+    if leetcode_url is not None:
```
Comment (khaimt) [inline run_inference.py]: if len(leetcode_url) > 0

File: run_inference.py [hunk 4]
```diff
@@ -1,31 +1,92 @@
...
+        algo_input = crawl_desc_leetcode_url(leetcode_url)
 
-    # read the leetcode problem from input_file
-    with open(input_file, "r") as f:
-        algo_input = f.read()
+    if algo_input is None:
```
Comment (khaimt) [inline run_inference.py:64]: if len(input_file) > 0

**Review by khaimt — APPROVED** [review]

**Pull request merged**